providers/provider_cdrom.go

```diff
@@ -28,7 +28,7 @@ import (
 // ListCDROMs lists all the cdroms in the system
 func ListCDROMs() []Provider {
 	// UserdataFiles is where to find the user data
-	var userdataFiles = []string{"user-data", "config"}
+	var userdataFiles = []string{"user-data", "meta-data", "config"}
```

Comment:
According to https://github.com/rancher/elemental-toolkit/issues/2125 both files (`user-data` and  `metadata`) are created for the Harvester use case. However the change here will only honor first match. I am not sure this is what we want. I am fine with the change proposed here to also check for `meta-data`, however the logic of this datasource only reads the first match (see snipped below), as soon as file is read stops trying the next one. So if both files are available at the same time we might need so more elaborated change.

@robertwbl what you think? does it make sense to you? I have the feeling this change is not enough after reading the issue you created and checking how this `userdataFiles` slice is being used.

https://github.com/rancher-sandbox/linuxkit/blob/fa64b2f3854601d59b3b85c3d227b64131259438/providers/cdrom_shared.go#L80-L101

Reply:
Ah, good spot - I hadn't read the usage clearly! You're right that this wouldn't work since Harvester provides both meta-data and user-data separately and simultaneously. I'll have a think about how this might work.

Reply:
I've just been exploring the original linuxkit repo this was forked from and it seems that has a way of handling the meta-data file: https://github.com/linuxkit/linuxkit/blob/be7dfdd42c6cc9365079f101b01536062729eda7/pkg/metadata/provider_cdrom.go#L121

Reply:
It looks as though in this fork, that behaviour got factored out in #1 . @davidcassany your commit message on https://github.com/rancher-sandbox/linuxkit/pull/1/commits/9106b9d218542bad51f71affec73d93dd3be2470 makes it sounds like at the time it was just considered to be unused and there's no other reason it was removed? I'd be keen to factor it back in this way if so.

Reply:
I've factored this back in in a way that also considers the open stack config drive provider. According to https://docs.openstack.org/nova/rocky/user/config-drive.html it can also be expected to find a `meta_data.json` file here. Let me know what you think, thanks!